strapi/strapi
# Make sizeLimit a plugin param and verify file size for every provider

### What does it do?

Allows users to specify a `sizeLimit` value in the `upload.config` object to limit the maximum size of uploaded files.

### Why is it needed?

Prior to this PR, `sizeLimit` was a provider option that could only be used with `upload-local`, but people might want to limit file sizes to reduce costs even when using other providers.

### How to test it?

Set a `sizeLimit` value (in bytes) in `upload.config` and notice that a `PayloadTooLargeError` is thrown if the file you are trying to upload is bigger than the threshold specified.

Note: if this PR is accepted, the Strapi docs should be updated to reflect the changes.
